inc/functions.inc.php

```diff
@@ -44,6 +44,85 @@ function st_register_widget() {
     }
 }
 
+function tp_disp_boolean($bool_text)
+{
+    $bool_text = (string)$bool_text;
+    if (empty($bool_text) || '0' === $bool_text || 'false' === $bool_text) {
+        return 'false';
+    }
+
+    return 'true';
+}
+
+/**
+ * Add custom filter on edit posts page
+ * 
+ * Filter posts by year
+ */
+
+//var_dump(get_option('taxopress_external_taxonomies')['category']['show_filters']);
+
+$show_filters = tp_disp_boolean(get_option('taxopress_external_taxonomies')['category']['show_filters']);
+//var_dump($show_filters);
+
+if ( $show_filters == 'true' ) {
+    add_action('restrict_manage_posts', 'taxopress_add_posts_filter');
+    add_filter( 'parse_query', 'taxopress_filter_by_year');
+} 
+
+
+
+function taxopress_add_posts_filter() {
+
+    global $wpdb, $table_prefix;
+    
+    $post_type = (isset($_GET['post_type'])) ? $_GET['post_type'] : 'post';
+
+    //only add filter to posts
+    if ($post_type == 'post'){
+        
+        //query database to get a list of years for the specific post type:
+        $values = array();
+        $query_years = $wpdb->get_results("SELECT year(post_date) as year from ".$table_prefix."posts
+                where post_type='".$post_type."'
+                group by year(post_date)
+                order by post_date");
+        foreach ($query_years as &$data){
+            $values[$data->year] = $data->year;
+        }
+
+        //give a unique name in the select field
+        ?><select name="tp_fiilter_year">
+<option value="">All years</option>
```

Comment:
@jonathanpaulbbaroma All strings should be translation ready. Hardcoding text without passing it through translation key will make it impossible for translator to find newly added strings when translating or making update to translation files.